README.md
# CIS227Group2
Hangman Game - Group 2


Change_Log.txt
Effective: 02.15.2022 8.30PM


Hangman.h
>merged files

ReadFile.H
>no change

ArrayBuilder.h
>no change

Main.cpp
>merged files
>edited Hint Y/N? to accept both uppercase and lowercase Y and N
>edited Hint Y/N? to keep asking for correct intput until a Y or N is entered

Group2_Hangman.h
#pragma once
/* File Name: Group2_Hangman.h
*    Authors: Johnson, Pelton, Simmons, Thompson
*       Date: 02/06/2022
*  Rev. Date: 02/21/2022
*    Version: 0.5.0
*/

//blueprints Hangman class

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include "Group2_Scorekeeper.h"         /*includeed scorekeeper.h*/

using namespace std;

class Hangman
{
private:
	string word;
	int badGuesses;
	unsigned int points = 0;
	bool win = false;
	bool foundLetter;
	bool hintGiven = false;
	bool goodGuess = false;
	vector<char> wordVector;
	vector<char> guessedLetters;
	
	Scorekeeper score;

public:

	//constructor
	Hangman()
	{
		badGuesses = 0;
		foundLetter = false;

		wordVector.clear();
		guessedLetters.clear();
	}

	//--------METHODS--------
	 
	//resets values on game end
	void GameClear()
	{
		badGuesses = 0;
		hintGiven = false;
		points = 0;

		wordVector.clear();
		guessedLetters.clear();

		score.clearScore();                               /*added the clearScore*/
	}

	void Guess(char letter)
	{
		goodGuess = false;
		
		//Section checks if input letter has been guessed already
		if (!guessedLetters.empty())
		{
			for (unsigned int j = 0; j < (guessedLetters.size()); j++)
			{
				if (tolower(guessedLetters[j]) == tolower(letter))
				{
					cout << "\nYou already guessed " << letter << "!" << endl;
					return;
				}
			}
		}

		//Section checks if letter is correct guess or not
		for (unsigned int i = 0; i < (wordVector.size()); i++)
		{
			if (tolower(wordVector[i]) == tolower(letter) && goodGuess != true)
			{
				cout << "\nYes, this word contains a(n): " << letter << endl;
				goodGuess = true;
				guessedLetters.push_back(tolower(letter));
				score.correctLetter();                                     /*add +1 pt for each correct letter guess*/
			}
		}

		//On a bad guess, moves badGuesses closer to a losing position
		if (!goodGuess)
		{
			badGuesses += 1;
			cout << "\nSorry! Try another letter" << endl;
			guessedLetters.push_back(letter);
		}

		BuildASCIIMan();
		DisplayWordSpaces();
		DisplayGuessedLetters();
		SetPoints();
		YouLose();
		YouWin();
	}

	void DisplayWordSpaces()
	{
		bool foundLetter;
		cout << endl;
		cout << "\n\t\t\t\t\t\t\t\t";

		//Goes through wordVector one letter at a time
		for (unsigned int i = 0; i < wordVector.size(); i++)
		{
			foundLetter = false;

			//checks guessedLetters against the above letter
			for (unsigned int j = 0; j < guessedLetters.size(); j++)
			{
				if (tolower(guessedLetters[j]) == tolower(wordVector[i]))
				{
					//if match, print letter
					foundLetter = true;
					cout << wordVector[i];	
				}

				if ((j == guessedLetters.size() - 1) && !foundLetter)
				{
					//if no matches, print _
					cout << "_";
				}
			}

			if (i == wordVector.size() - 1)
			{
				//when done, print new line
				cout << endl;
			}
		}
	}

	//section displays each guessed letter
	void DisplayGuessedLetters()
	{
		cout << "\nGuessed Letters :";
		for (unsigned int j = 0; j < (guessedLetters.size()); j++)
		{
			cout << " " << guessedLetters[j];
		}
	}

	//Section displays the hangman character based on number of bad guesses
	void BuildASCIIMan()
	{
		switch (badGuesses)
		{
		case 1:
			cout << "\n\t\t\t\t\t\t\t\t      _______  \n";
			cout << "\t\t\t\t\t\t\t\t      ;     | \n";
			cout << "\t\t\t\t\t\t\t\t      O     | \n";
			cout << "\t\t\t\t\t\t\t\t            | \n";
			cout << "\t\t\t\t\t\t\t\t            | \n";
			break;

		case 2:
			cout << "\n\t\t\t\t\t\t\t\t      _______  \n";
			cout << "\t\t\t\t\t\t\t\t      ;     | \n";
			cout << "\t\t\t\t\t\t\t\t      O     | \n";
			cout << "\t\t\t\t\t\t\t\t      |     | \n";
			cout << "\t\t\t\t\t\t\t\t            | \n";
			break;

		case 3:
			cout << "\n\t\t\t\t\t\t\t\t      _______  \n";
			cout << "\t\t\t\t\t\t\t\t      ;     | \n";
			cout << "\t\t\t\t\t\t\t\t      O     | \n";
			cout << "\t\t\t\t\t\t\t\t     /|     | \n";
			cout << "\t\t\t\t\t\t\t\t            | \n";
			break;

		case 4:
			cout << "\n\t\t\t\t\t\t\t\t      _______  \n";
			cout << "\t\t\t\t\t\t\t\t      ;     | \n";
			cout << "\t\t\t\t\t\t\t\t      O     | \n";
			cout << "\t\t\t\t\t\t\t\t     /|\\    | \n";
			cout << "\t\t\t\t\t\t\t\t            | \n";
			break;

		case 5:
			cout << "\n\t\t\t\t\t\t\t\t      _______  \n";
			cout << "\t\t\t\t\t\t\t\t      ;     | \n";
			cout << "\t\t\t\t\t\t\t\t      O     | \n";
			cout << "\t\t\t\t\t\t\t\t     /|\\    | \n";
			cout << "\t\t\t\t\t\t\t\t     /      | \n";
			break;

		case 6:
			cout << "\n\t\t\t\t\t\t\t\t      _______  \n";
			cout << "\t\t\t\t\t\t\t\t      ;     | \n";
			cout << "\t\t\t\t\t\t\t\t      O     | \n";
			cout << "\t\t\t\t\t\t\t\t     /|\\    | \n";
			cout << "\t\t\t\t\t\t\t\t     / \\    | \n";
			break;

		default:
			break;
		}
	}

	void Hint()
	{
		//reset value
		foundLetter = false;
		char letter;

		do
		{
			for (unsigned int i = 0; i < wordVector.size(); i++)
			{
				foundLetter = false;
				//This will become the given hint, if not in guessedLetters
				letter = wordVector[i];

				for (unsigned int j = 0; j < guessedLetters.size(); j++)
				{
					if ((tolower(guessedLetters[j]) == tolower(letter)))
					{
						foundLetter = true;
						break;
					}
					else if (!foundLetter)
					{
						guessedLetters.push_back(tolower(letter));
						//flag to disable further hints and exit loop
						hintGiven = true;
						break;
					}
				}

				if (hintGiven)
				{
					break;
				}
			}
		} while (!hintGiven);
	}

	bool YouLose()
	{
		//Bad guesses cause a losing condition
		if (badGuesses < 6)
		{
			return true;
		}

		win = false;
		return false;
	}

	void SetPoints()
	{
		//Section compares each letter in wordVector to guessed letters
		for (char wordVectorItem : wordVector)
		{
			if (tolower(wordVectorItem) == tolower(guessedLetters[guessedLetters.size()-1]))
			{
				//On match, add a point.
				points += 1;
			}
		}
		//extra points for guessing the entire word
		if (points >= wordVector.size())
		{
			points += 5;
		}
	}

	bool YouWin()
	{
		//Get points equal to the length of wordVector to win
		if (points >= wordVector.size())
		{
			win = true;
			return true;
		}

		return false;
	}


	bool GetWin()
	{
		return win;
	}

	//destructor
	~Hangman()
	{
		//empty
	}

	//setter
	void SetHangmanWord(string _word)
	{
		word = _word;
		copy(word.begin(), word.end(), back_inserter(wordVector));
	}

	//getters
	string GetHangmanWord()
	{
		return word;
	}

	int GetBadGuesses()
	{
		return badGuesses;
	}

	short unsigned int GetPoints()
	{
		return points;
	}

	bool GetHintGiven()
	{
		return hintGiven;
	}

	void gameWin()                                   /*added methods to call upon winning or losing*/
	{
		score.correctWord();
		cout << "\n\n\nWINNER! Good Job!" << endl;
		cout << "Score: " << score.outputScore() << endl;
	}

	void gameLose()
	{
		cout << "\n\nSorry! The word was: " << GetHangmanWord() << endl;
		cout << "\nScore: " << score.outputScore() << endl;
		cout << "\nBetter luck next time!" << endl;
	}

	short int gameTotalScore()                         /*added method to output Scorekeeper*/
	{
		short int result = score.outputScore();
		return result;
	}
};


Group2_Players.h
/* File Name: Group2_Players.cpp
*    Authors: Johnson, Pelton, Simmons, Thompson
*	    Date: 02/21/2022
*  Rev. Date: 02/22/2022
*    Version: 0.5.3
*/

#include <map>
#include <fstream>
#include <iostream>

using namespace std;

class Player
{
    private:
        string username;
        unsigned short int points;
        multimap<int, string, std::greater<int> > leaderboard;

    public:

        //constructor
        Player()
        {
            username = "Player";
            points = 0;
        }

        //set and get username
        void SetPlayerName(string name)
        {            
            username = name;
        }

        string GetPlayerName() {
            return username;
        }

        //set and get points
        void SetPoints(int value)
        {
            points += value;
        }

        unsigned short int GetPoints()
        {
            return points;
        }


        //-----LEADERBOARD METHODS-----
        void UpdateLeaderboard()
        {
            leaderboard.insert(make_pair(points, username));
        }
        
        void ResetPlayer()
        {
            username = "Player";
            points = 0;
        }

        void PrintLeaderboard()
        {
            short unsigned int i = 1;

            multimap<int, string>::iterator LeaderboardIterator = leaderboard.begin();

            cout << "--------Leaderboard--------" << endl;

            while (LeaderboardIterator != leaderboard.end())
            {
                //prints Top Ten Leaderboard             
                cout << i << " - " << LeaderboardIterator->second << ": ";
                cout << LeaderboardIterator->first << " Points" << endl;
                i++;
                LeaderboardIterator++;

                if (i == 11)
                {
                    break;
                }

            }
        }
};

Group2_ReadFile.h
#pragma once
/* File Name: Group2_ReadFile.h
*    Authors: Johnson, Pelton, Simmons, Thompson
*       Date: 02/03/2022
*  Rev. Date: 02/10/2022
*    Version: 0.5.0
*/

//blueprints ReadFileToArray class
//COMPLETE: TEMPORARILY change vector to list
//ERROR: if there is [spacebar] in file, reads the [spacebars] as part of the word

#include <fstream>
#include <vector>
#include <list>

class ReadFileToArray
{
private:
	//std::vector<std::string> strArray;
	std::list<std::string> strArray;

public:
	ReadFileToArray()
	{
		//default constructor, no paramater
		std::string strLine;
		std::ifstream txtFile;

		txtFile.open("Group2_TxtFile.txt");

		if (txtFile.is_open())
		{
			while (getline(txtFile, strLine))
			{
				unsigned short int index = 0;

				//trim strLine of whitespace at start
				//strLine.erase(std::remove(strLine.begin(), strLine.end(), ' '), strLine.end());

				strArray.push_back(strLine);
				
				++index;
			}
		}
		txtFile.close();
	}

	~ReadFileToArray()
	{
		//destructor
	}

	//outputs one line of vector
	std::string outputLineOfFile(int x)
	{
		//std::string line = strArray[x];

		std::list<std::string>::iterator it = strArray.begin();
		for (int i = 0; i < x; i++) {
			++it;
		}

		std::string line = *it;

		return line;
	}
	
	//outputs whole vector
	std::list<std::string> outputReadFileToArray()
	{
		return strArray;
	}

};


Group2_Players.cpp
/* File Name: Group2_Players.cpp
*    Authors: Johnson, Pelton, Simmons, Thompson
*	    Date: 02/19/2022
*    Version: 0.5.2
*/

#include <map>
#include <fstream>
#include <iostream>

using namespace std;

class Player
{
    private:
        string username;
        unsigned short int points;
        multimap<string, int> leaderboard;

    public:

        //constructor
        Player()
        {
            username = "Player";
            points = 0;
        }

        //set and get username
        void SetPlayerName(string name)
        {
            username = name;
        }

        string GetPlayerName() {
            return username;
        }

        //set and get points
        void SetPoints(int value)
        {
            points += value;
        }

        unsigned short int GetPoints()
        {
            return points;
        }


        //-----LEADERBOARD METHODS-----
        void UpdateLeaderboard()
        {
            leaderboard.insert(make_pair(username, points));
        }
        
        void ResetPlayer()
        {
            username = "Player";
            points = 0;
        }

        void PrintLeaderboard()
        {
            short unsigned int num = -1;
            short unsigned int i = 1;

            multimap<string, int>::iterator LeaderboardIterator = leaderboard.begin();

            cout << "--------Leaderboard--------" << endl;

            while (LeaderboardIterator != leaderboard.end())
            {
                cout << i << " - " << LeaderboardIterator->first << ": ";
                cout << LeaderboardIterator->second << " Points" << endl;
                i++;
                LeaderboardIterator++;

                if (i == 10)
                {
                    break;
                }

            }
        }
};

Group2_ArrayBuilder.h
#pragma once
/* File Name: Group2_ArrayBuilder.h
*    Authors: Johnson, Pelton, Simmons, Thompson
*       Date: 01/30/2022
*  Rev. Date: 02/08/2022
*    Version: 0.5.0
*/

//blueprints ArrayBuilder class
//POSSIBLE: convert to using vector (instead of array)

using namespace std;

class ArrayBuilder
{
private:
	string strArray[12];

public:
	ArrayBuilder()
	{
		//constructor
		for (unsigned short int i = 0; i < 12; i++)
		{
			strArray[i] = "placeholder" + to_string(i + 1);
		}
	}
	~ArrayBuilder()
	{
		//destructor
	}
	void SetArrayBuilderValue(std::string& str, int x)
	{
		//setter
		strArray[x] = str;
	}
	string GetArrayBuilderValue(int x)
	{
		//getter
		return strArray[x];
	}

	void printArrayBuilder()
	{
		for (unsigned short int i = 0; i < 12; i++)
		{
			cout << strArray[i] << endl;
		}
	}
};


Group2_Main.cpp
/* File Name: Group2_Main.cpp
*    Authors: Johnson, Pelton, Simmons, Thompson
*	    Date: 01/30/2022
*  Rev. Date: 02/21/2022
*    Version: 0.5.4
*/

//POSSIBLE: convert all arrays to vectors
//ERROR: when entering new userName, cannot have spaces


#include <iostream>
#include <fstream>
#include <string>
#include <array>
#include <vector>
#include <list>
#include <map>
#include "Group2_ArrayBuilder.h"
#include "Group2_ReadFile.h"
#include "Group2_Hangman.h"
#include "Group2_Leaderboard.h"
#include "Group2_Players.h"

using namespace std;

int main(int argc, char* argv[])
{

	//declaring variables
	bool mainMenu = true;
	bool optionMenu = true;
	bool exitHangman = false;

	string versionNumber = "0.5.4";
	string input;
	string word;
	string userName = "Player";
	unsigned short int choice = 0;
	short int option;
	short int score;
	ArrayBuilder arrayClass;
	ReadFileToArray txtFile;
	Hangman game;
	Player player;


	//command line argument handling
	//cout << "--------Command Line Arguments--------" << endl;
	//cout << "Number of Arguments " << argc << endl;

	//Checks for the information command line argument
	if ((argc > 1) && (string)argv[1] == "-i")
	{
		cout << "\n\nArray Assignment";
		cout << "\nVersion " + versionNumber;
		cout << "\nCreated by Johnson, Pelton, Simmons, and Thompson\n\n";
	}

	if (argc <= 13)
	{
		//transferring argv[] to array from ArrayBuilder class
		cout << "Transferring Arguments to New List . . ." << endl;

		for (unsigned short int i = 0; i < (argc - 1); i++) {
			string temp = string(argv[i + 1]);
			
			//ignores the -i command
			if (temp != "-i")
			{
				arrayClass.SetArrayBuilderValue(temp, i);
			}
		}
		cout << "Transfer Complete" << endl;
	}


	//commands and options, until exit (-e) is selected
	do
	{
		//commands main menu
		cout << "\n--------Commands:--------\n";
		cout << "Information:     -i" << endl;
		cout << "Options:         -o" << endl;
		cout << "Exit:            -e" << endl;

		cin >> input;
		if (input == "-i")
		{
			//information command
			cout << "\n--------Information:--------\n";
			cout << "Program Name:       Array Builder\n";
			cout << "Authors:            Johnson, Pelton, Simmons, Thompson\n";
			cout << "Description:        Initially creates an array from command line arguments.\n";
			cout << "                    Offers the user options to manipulate, update, or view the array.\n";
			cout << "                    Offers the user an option to play Hangman with a selected word.\n";
			cout << "Version:            " + versionNumber << endl;
		}
		else if (input == "-o")
		{
			do
			{
				//options command; options sub-menu
				cout << "\n--------Options:--------\n";
				cout << "1 - Update List with Console\n";
				cout << "2 - Update List with Text File\n";
				cout << "3 - Display a Word\n";
				cout << "4 - Update One Word\n";
				cout << "5 - Print List\n";
				cout << "6 - Guess the Letters (Hangman!)\n";
				cout << "7 - View Leaderboard\n";
				cout << "8 - Return to Commands\n";

				cin >> input;
				//testing if input is an integer
				try
				{
					choice = stoi(input);
					//doesn't catch #aaaa
				}
				catch (...)
				{
					choice = 0;
				}

				//case switch with program options
				switch (choice)
				{
				case 1:
					//enter new words for entire list with console
					for (unsigned short int i = 0; i < 12; i++)
					{
						cout << "\nInput word for #" << i + 1 << "\n";
						cin >> input;
						arrayClass.SetArrayBuilderValue(input, i);
					}
					cout << "List Updated\n";

					break;

				case 2:
					//overwrite list with txt file
					cout << "\nOverwriting List with Text File . . .\n";

					for (unsigned short int i = 0; i < 12; i++)
					{
						string temp = txtFile.outputLineOfFile(i);
						arrayClass.SetArrayBuilderValue(temp, i);
					}
					cout << "\nOverwrite Complete\n";

					break;

				case 3:
					//displays single word
					cout << "\nWhich word would you like to display? Input a number 1 to 12:\n";
					cin >> input;

					//checks if input is an integer
					try
					{
						option = stoi(input);
						//doesn't catch #aaaa
					}
					catch (...)
					{
						cout << "\nIncorrect input. Default option selected: 1\n";
						option = 1;
					}

					//checks if option is between 1 and 12
					if ((option >= 1) && (option <= 12))
					{
						cout << arrayClass.GetArrayBuilderValue(option - 1) << endl;
					}
					else
					{
						cout << "\nIncorrect input. Default option selected: 1\n";
						cout << arrayClass.GetArrayBuilderValue(0) << endl;
					}
					option = -1;
					break;

				case 4:
					//update one word
					cout << "\nWhich word to update? Input a number 1 to 12:\n";
					cin >> input;

					//checks if input is an integer
					try
					{
						option = stoi(input);
						//doesn't catch #aaaa
					}
					catch (...)
					{
						cout << "\nIncorrect input. Default option selected: 1\n";
						option = 1;
					}

					//checks if option is between 1 and 12
					if ((option >= 1) && (option <= 12))
					{
						//continue
					}
					else
					{
						cout << "\nIncorrect input. Default option selected: 1\n";
						option = 1;
					}

					cout << "\nEnter a word:\n";
					cin >> input;
					arrayClass.SetArrayBuilderValue(input, option - 1);
					cout << "\nList Updated\n";

					break;

				case 5:
					//prints list
					cout << "\nPrinting List:\n" << endl;
					arrayClass.printArrayBuilder();
					break;

				case 6:
					//Guess the Letters (Hangman)
					cout << "\n--------HANGMAN--------" << endl;
					
					//sets userName
					cout << "\nCurrent player: " << userName << endl;
					cout << "Would you like to change player? (Y/N)" << endl;
					cin >> input;
					do
					{
						if (input == "Y" || input == "y")
						{
							cout << "\nPlease enter your name: ";
							cin >> userName;

							break;
						}
						else if (input == "N" || input == "n")
						{
							cout << "\nProceeding as " << userName << endl;
							
							break;
						}
						else
						{
							cout << "\nIncorrect Input" << endl;
							cout << "\nCurrent player: " << userName << endl;
							cout << "Would you like to change player? (Y/N)" << endl;
							cin >> input;
						}
					} while ( (input == "Y") || (input == "N") || (input == "y") || (input == "n"));

					//starts game with chosen userName
					player.SetPlayerName(userName);
					cout << "\nWelcome, " << player.GetPlayerName() << "!" << endl;
					
					//player continuously plays game until exit
					do
					{
						cout << endl;
						cout << userName << ", please pick a number from 1 to 12: ";
						cin >> input;
						//checks if input is an integer
						try
						{
							option = stoi(input);
							//doesn't catch #aaaa
						}
						catch (...)
						{
							cout << "\nIncorrect input. Default option selected: 1\n";
							option = 1;
						}

						//Selects word to play Hangman with
						if ((option >= 1) && (option <= 12))
						{
							word = arrayClass.GetArrayBuilderValue(option - 1);
						}
						else
						{
							cout << "\nIncorrect input. Default option selected: 1\n";
							word = arrayClass.GetArrayBuilderValue(0);
						}
						//Resets option value
						option = -1;

						cout << endl;
						cout << userName << ", input a single letter as a guess when prompted, and try not to get too many wrong guesses!\n\n";
						game.SetHangmanWord(word);

						do
						{
							if (game.GetBadGuesses() > 3 && game.GetHintGiven() == false)
							{
								cout << "\nNeed a hint? (Y/N)" << endl;
								cin >> input;

								bool hintYN = false;
								do
								{
									if (input == "Y" || input == "y")
									{
										game.Hint();
										cout << "\nHere's a free letter, " << userName << "! \nYou only get one, so guess wisely...\n\n" << endl;
										game.DisplayWordSpaces();
										hintYN = true;
									}
									else if (input == "N" || input == "n")
									{
										cout << "\nOkay, no hint yet!" << endl;
										hintYN = true;
									}
									else
									{
										cout << "\nIncorrect Input" << endl;
										cout << "\nNeed a hint? (Y/N)" << endl;
										cin >> input;
									}
								} while (!hintYN);
							}

							cout << "\nGuess a letter!" << endl;
							cin >> input;
							//Only grabs first letter of input
							game.Guess(input[0]);

						} while (!game.YouWin() && game.YouLose());

						if (game.GetWin())
						{
							game.gameWin();
							player.SetPoints(game.gameTotalScore());
							game.GameClear();

							//Continue? (Y/N)
							cout << "\nWould you like to play another round, " << player.GetPlayerName() << "? (Y/N): ";
							cin >> input;
							if (input == "Y" || input == "y")
							{
								cout << "\nOkay, let's play again!" << endl << endl;
							}
							else if (input == "N" || input == "n")
							{
								cout << "\nThanks for playing, " << player.GetPlayerName() << "!" << endl << endl;
								exitHangman = true;
							}

							else
							{
								cout << "\nIncorrect Input" << endl;
								cout << "\nWould you like to play another round, " << player.GetPlayerName() << "? (Y/N): ";

								cin >> input;
							}
						}
						else
						{
							game.gameLose();
							player.SetPoints(game.gameTotalScore());
							game.GameClear();

							//Continue? (Y/N)
							cout << "\nWould you like to play another round, " << player.GetPlayerName() << "? (Y/N): ";
							cin >> input;
							if (input == "Y" || input == "y")
							{
								cout << "\nOkay, let's play again!" << endl << endl;
							}
							else if (input == "N" || input == "n")
							{
								cout << "\nThanks for playing, " << player.GetPlayerName() << "!" << endl << endl;
								exitHangman = true;
							}

							else
							{
								cout << "\nIncorrect Input" << endl;
								cout << "\nWould you like to play another round, " << player.GetPlayerName() << "? (Y/N): ";

								cin >> input;
							}
						}
					} while (!exitHangman);

					exitHangman = false;
					player.UpdateLeaderboard();
					player.ResetPlayer();
					break;
            
				case 7:
					//prints leaderboard
					player.PrintLeaderboard();
					break;
            
				case 8:
					//exits program
					cout << "\nExiting Options . . ." << endl;
					optionMenu = false;
					break;

				default:
					cout << "\nInvalid Input." << endl;
					break;
				}
			} while (optionMenu);
		}
		else if (input == "-e")
		{
			//exit command
			cout << "\nExiting Program . . ." << endl;
			mainMenu = false;
		}
		else
		{
			cout << "Invalid input." << endl << endl;
		}
	} while (mainMenu);


	return 0;
};
